packages/babel-plugin-ember-test-metadata/get-normalized-file-path.js

```diff
@@ -1,36 +1,32 @@
-const {
-  _getRelativePathForClassic,
-  _getRelativePathForClassicInRepo,
-  _getRelativePathForEmbroider,
-  _getRelativePathForEmbroiderInRepo,
-} = require('./get-relative-paths');
-
-const path = require('path');
-
 /**
  * Get a normalized file path, based on whether the app build is classic or with Embroider
  * @param {object} opts Babel state.file.opts which include root and filename props
  * @returns {string} E.g. tests/acceptance/my-test.js
  */
-function getNormalizedFilePath({ packageName, isUsingEmbroider, filename, root }) {
-  if (!isUsingEmbroider) {
-    if (filename.includes('ember-add-in-repo-tests')) {
-      return _getRelativePathForClassicInRepo(filename);
-    }
+function getNormalizedFilePath({ packageName, filename, isUsingEmbroider, root, sourceFileName }) {
+  let file = sourceFileName || filename;
 
-    return _getRelativePathForClassic(filename, packageName);
-  } else {
-    const rootDirWithBase = path.join(path.parse(root).dir, path.parse(root).base);
-    if (filename.includes(rootDirWithBase)) {
-      filename = filename.replace(rootDirWithBase, '');
-    }
+  if (file.includes(root)) {
+    const regex = new RegExp(`.*${root}/`);
+    file = file.replace(regex, '');
+  }
 
-    if (filename.includes('ember-add-in-repo-tests')) {
-      return _getRelativePathForEmbroiderInRepo(filename);
-    }
+  if (file.startsWith(`${packageName}/`)) {
+    const regex = new RegExp(`^${packageName}/`);
+    file = file.replace(regex, '');
+  }
 
-    return _getRelativePathForEmbroider(filename);
+  if (isUsingEmbroider) {
+    const regex = new RegExp(`^.*embroider/.{6}/`);
+    file = file.replace(regex, '');
   }
+
+  if (file.includes('ember-add-in-repo-tests')) {
+    const regex = new RegExp(`^.*ember-add-in-repo-tests/`);
+    file = file.replace(regex, '');
+  }
+
+  return file;
```

Comment:
I think this results in code that is harder to read. The reason we split the different permutations out into well factored units was so that we could encapsulate the logic into semantically named functions, which makes it much easier to go back through and understand the intent of each. Moving them all into inline ifs defeats that purpose.

If I'm not mistaken, this was an early direction that was pursued only to be discarded in favor of the well factored units into functions approach.

Additionally, it made testing these permutations discrete, and readable. We've also removed a lot of the tests that had provided us value through quick iteration.

I'm not in favor of us continuing down this path.

Reply:
thanks for the input. Discrete functions and tests were be helpful in ironing out the implementation and building confidence in our design choices. That said, we have the same cases covered in `index-test.js`. This approach made sense to me because there's far less indirection, everything you want to know about the transformations are in one place. what are your thoughts?

Reply:
I could also split out these blocks into individual helpers in the same file, keeping the newer implementation and readding the unit tests.. hows that sound?

Reply:
IMO the tests are effective in describing the behavior, but that's easy enough to split out utility methods for those blocks if you think it's easier to maintain and read. I'm reverting to the original design, but this code works right now.